clients/src/main/java/org/apache/kafka/clients/consumer/internals/FetchRequestManager.java

```diff
@@ -0,0 +1,136 @@
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import org.apache.kafka.clients.ClientResponse;
+import org.apache.kafka.clients.FetchSessionHandler;
+import org.apache.kafka.clients.consumer.internals.NetworkClientDelegate.PollResult;
+import org.apache.kafka.clients.consumer.internals.NetworkClientDelegate.UnsentRequest;
+import org.apache.kafka.common.Node;
+import org.apache.kafka.common.requests.FetchRequest;
+import org.apache.kafka.common.utils.LogContext;
+import org.apache.kafka.common.utils.Time;
+import org.slf4j.Logger;
+
+/**
+ * {@code FetchRequestManager} is responsible for generating {@link FetchRequest} that represent the
+ * {@link SubscriptionState#fetchablePartitions(Predicate)} based on the user's topic subscription/partition
+ * assignment.
+ */
+public class FetchRequestManager extends AbstractFetch implements RequestManager {
+
+    private final Logger log;
+    private final ErrorEventHandler errorEventHandler;
+    private final NetworkClientDelegate networkClientDelegate;
+
+    FetchRequestManager(final LogContext logContext,
+                        final Time time,
+                        final ErrorEventHandler errorEventHandler,
+                        final ConsumerMetadata metadata,
+                        final SubscriptionState subscriptions,
+                        final FetchConfig fetchConfig,
+                        final FetchMetricsManager metricsManager,
+                        final NetworkClientDelegate networkClientDelegate) {
+        super(logContext, metadata, subscriptions, fetchConfig, metricsManager, time);
+        this.log = logContext.logger(FetchRequestManager.class);
+        this.errorEventHandler = errorEventHandler;
+        this.networkClientDelegate = networkClientDelegate;
+    }
+
+    @Override
+    protected boolean isUnavailable(Node node) {
+        return networkClientDelegate.isUnavailable(node);
+    }
+
+    @Override
+    protected void maybeThrowAuthFailure(Node node) {
+        networkClientDelegate.maybeThrowAuthFailure(node);
+    }
+
+    @Override
+    public PollResult poll(long currentTimeMs) {
+        List<UnsentRequest> requests;
+
+        if (!idempotentCloser.isClosed()) {
+            // If the fetcher is open (i.e. not closed), we will issue the normal fetch requests
+            requests = prepareFetchRequests().entrySet().stream().map(entry -> {
+                final Node fetchTarget = entry.getKey();
+                final FetchSessionHandler.FetchRequestData data = entry.getValue();
+                final FetchRequest.Builder request = createFetchRequest(fetchTarget, data);
+                final BiConsumer<ClientResponse, Throwable> responseHandler = (clientResponse, t) -> {
+                    if (t != null) {
+                        handleFetchResponse(fetchTarget, t);
+                        log.warn("Attempt to fetch data from node {} failed due to fatal exception", fetchTarget, t);
```

Comment:
it is handled in the error event handler - which means the user should get the exception upon invoking poll. I wonder if we could just log info or at a different level. or even no logging.

Reply:
Are you saying that the incoming error (`t`) is already logged before the callback is invoked? If so, then yeah, it makes sense to remove it here.

Reply:
It looks like the other call sites that add error events to the handler also log them 🤷‍♂️